canonical-web-and-design/ubuntu.com
# Bulk api

## Done

- One endpoint to bulk create and update
- Refactor (old endpoints and error handling)
- Fixed issues with main table.

## QA

- Check out this feature branch
- Delete the `usn.sqlite3` file
- Run the site using the command `./run serve` or `dotrun`
- Run `python webapp/security/fixtures/releases.py`
- Download/clone/rebase current ubuntu [tracker upload script](https://github.com/nottrobin/ubuntu-cve-tracker/pull/2) 
- Run  `python scripts/upload-cve.py .../active` in ubuntu tracker, it will upload the last 100 CVEs, but this order could be random depending on filesystem order, so you could not be grabbing the latest CVE. Print the `list_cves` in the upload script if you want to get an accurate CVE list you are sending
- You can check `python scripts/upload-cve.py --help` to remind you of the possible parameters.
- Check `http:localhost:8001/security/cve`, if you can't find it (because 100s of CVEs have been uploaded) check - `> sqlite3 SELECT id FROM cve WHERE id={CVE-id}` where CVE-id is a CVE you uploaded with the script. 

## Issue / Card

Fixes #7302
